File: Data Structures & Algorithms/car-fleet/submission-1.py
class Solution:
    def carFleet(self, target: int, position: List[int], speed: List[int]) -> int:
        
        time_taken = [math.ceil((target - position[i]) / speed[i]) for i in range(len(position))]
        cum_list = [(time_taken[i], position[i]) for i in range(len(position))]
        cum_list.sort(key=lambda t: t[1])
        result = 0
        while cum_list: 
            curr = cum_list.pop()
            while cum_list and cum_list[-1] <= curr:
                cum_list.pop()
            result+=1
        
        return result

File: Data Structures & Algorithms/combination-target-sum-ii/submission-1.js
class Solution {
    /**
     * @param {number[]} candidates
     * @param {number} target
     * @return {number[][]}
     */
    combinationSum2(candidates, target) {
        let result = []; 
        candidates = candidates.sort(); 

        const bt = (start, combo, currSum) =>{
            if(currSum == target){
                result.push([...combo])
                return; 
            }
            if(currSum > target) return; 


            for(let i = start; i < candidates.length; i++){
                if( i > start && candidates[i] == candidates[i - 1]) continue
                combo.push(candidates[i]);
                bt(i + 1, combo, currSum + candidates[i]);
                combo.pop()
            }
        }

        bt(0, [], 0)
        return result; 
    }
}
